src/app-account.js

```diff
@@ -59,6 +60,10 @@ app.post('/login', async (req, res) => {
   let loginMethod = getLoginMethod(req);
   console.log('Logging in via ' + loginMethod);
   let tokenRes = null;
+  if (!config.allowedLoginMethods.includes(loginMethod)) {
```

Comment:
This is a security improvement. `getLoginMethod` prefers whatever's in the clientSide request. This allows us to reject that if the server admin has decided they don't want header auth enabled.